apps/checkout/src/hooks/useFetch.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from "react";
+
+export const useFetch = <TArgs extends Record<string, any>>(
+  fetchFn: (args?: TArgs) => Promise<any>,
+  args?: TArgs
+) => {
```

Comment:
Add a third argument called `opts`, with `skip` - this will not trigger fetch on useEffect.